src/Microsoft.DotNet.Darc/src/DarcLib/VirtualMonoRepo/VmrBinaryFileScanner.cs

```diff
@@ -0,0 +1,56 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.DotNet.Darc.Models.VirtualMonoRepo;
+using Microsoft.DotNet.DarcLib.Helpers;
+using Microsoft.Extensions.Logging;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+#nullable enable
+namespace Microsoft.DotNet.DarcLib.VirtualMonoRepo;
+
+public class VmrBinaryFileScanner : VmrScanner
+{
+    // Git output from the diff --numstat command, when it finds a binary file
+    private const string BinaryFileMarker = "-\t-";
+
+    public VmrBinaryFileScanner(
+        IVmrDependencyTracker dependencyTracker, 
+        IProcessManager processManager,
+        IVmrInfo vmrInfo, 
+        ILogger<VmrScanner> logger) 
+        : base(dependencyTracker, processManager, vmrInfo, logger)
+    {
+    }
+
+    protected override async Task<IEnumerable<string>> ScanRepository(SourceMapping sourceMapping, CancellationToken cancellationToken)
+    {
+        cancellationToken.ThrowIfCancellationRequested();
+        var args = new string[]
+        {
+            "diff",
+            Constants.EmptyGitObject,
+            "--numstat",
+            GetExclusionFilter(_vmrInfo.GetRepoSourcesPath(sourceMapping))
+        };
+
+        var ret = await _processManager.ExecuteGit(_vmrInfo.VmrPath, args.ToArray(), cancellationToken);
+
+        ret.ThrowIfFailed($"Failed to scan the {sourceMapping.Name} repository");
+
+        return ret.StandardOutput
+            .Split(Environment.NewLine, StringSplitOptions.RemoveEmptyEntries)
+            .Where(line => line.StartsWith(BinaryFileMarker))
+            .Select(line => line.Split('\t').Last());
+    }
+
+    protected override string ScanType() => "binary";
```

Comment:
If you want to keep this for the sake of the logging, just make it a property then? Also make sure to not create the string over and over.

`protected override string ScanType { get; } = "binary";`

Reply:
done